### DISCUSSION TITLE
weird naming convention for ESP32-s3 ADC for pitch bending

### DISCUSSION THREAD

**Discussion by josehdx** [tttapa/Control-Surface#849] [General]
Can anyone give me a hint about how to name a PB (ADC) following the attached picture?

I have an old esp32 (ttgo) where the ADC used is the #12 (right side of the screenshoot)  hence i used ths number in the code  and it works fine.

Then on my new esp32-s3 (pic on the left side) the naming is a litle weird. I have checked the S3 documentation and i read it says some naming like ADC2_CH6 (this is the pin that u want to use) but it does not seem to work

I wonder if this is because the library is not ready for the new esp32 boards or if because i am using a wrong number

![Screenshot 2022-11-17 215207](https://user-images.githubusercontent.com/22876366/202559033-319d2a03-2e0d-4989-b4bc-417395767e48.jpg)

Thank you

**Comment by josehdx** [comment]
Just to clarify, the chanels the board on the left is talking about, are not related to the MIDI channels of the code

**Comment by tttapa** [comment]
Control Surface doesn't change the pin numbers, it just calls `analogRead(pinNumber)` under the hood, where `pinNumber` is whatever pin you give it. 

I believe you should specify the GPIO number instead of the number after `ADC` (i.e. 2 in this case, not 12). See https://docs.espressif.com/projects/arduino-esp32/en/latest/api/adc.html.

**Reply by josehdx** [reply]
ahhhh the GPIO was the solution :D Thanks

I have another doubt about a previous topic that was handle on "forcing middle position" on the PB function but it was done with a Physicall normal know.

I have a self centered (spring loaed) potentiometer, i have even printed on 3D some kind of mechanical Cam to it aplace on it a whammy bar so it will be a bluetooth controller for guitar

When i use the PB function along with this Potentiometer (https://tocos.com/products/series/mobility-scooter-parts/rvq28ys) it is not returning to the middle zone in the software. It happens in both direccion, Moving to the left, it returns to 48 % and moving to the right, sometimes returns to 48 and sometimes to 45%. Checking such values makes me thing, i should "move" the range of the mapping between 2 or 3 percent to the clockwise direction, hence the final position will be 50%

Also, when you mentioned in that topic this : ([0, FilteredAnalog<>::getMaxRawValue()]  my undestanding is that you are actually setting a virtual dead zone. But but my goal is to move the whole range in to make it fit to my software.

The thing to consider is that the same potentiometer has some eletric dead zones in the middle, around 8 degrees, 4 degrees per side.  (see picture) so i am not sure how could this dead zone can affect my goal as well. 

Do you think that just moving the "reading" zone via code  is a good idea? or would you suggest something else?


![Screenshot 2022-11-18 005516](https://user-images.githubusercontent.com/22876366/202585110-9ad41965-98bc-4f9e-b7fb-41b25ea8a1e1.jpg)

**Reply by tttapa** [reply]
> Do you think that just moving the "reading" zone via code is a good idea? or would you suggest something else?

Sure, a couple of `if` statements and some linear equations should do the trick.
